src/topology/sheaves/skyscraper.lean

```diff
@@ -0,0 +1,184 @@
+{ X := S,
+  ι := { app := λ U, eq_to_hom $ if_pos $ h.mem_open U.unop.1.2 U.unop.2,
+    naturality' := λ U V inc, begin
+      change dite _ _ _ ≫ _ = _, rw dif_pos,
+      { erw [category.comp_id, eq_to_hom_trans], refl },
+      { exact h.mem_open V.unop.1.2 V.unop.2 },
+    end } }
+
+/--
+The cocone at `S` for the stalk functor of `skyscraper_presheaf p₀ S` when `y ∈ closure {p₀}` is a
+colimit
+-/
+noncomputable def skyscraper_presheaf_cocone_is_colimit_of_specializes
+  {y : X} (h : p₀ ⤳ y) : is_colimit (skyscraper_presheaf_cocone_of_specializes p₀ S h) :=
+{ desc := λ c, eq_to_hom (if_pos trivial).symm ≫ c.ι.app (op ⊤),
+  fac' := λ c U, begin
+    rw ← c.w (hom_of_le $ (le_top : unop U ≤ _)).op,
+    change _ ≫ _ ≫ dite _ _ _ ≫ _ = _,
+    rw dif_pos,
+    { simpa only [skyscraper_presheaf_cocone_of_specializes_ι_app,
+        eq_to_hom_trans_assoc, eq_to_hom_refl, category.id_comp] },
+    { exact h.mem_open U.unop.1.2 U.unop.2 },
+  end,
+  uniq' := λ c f h, by rw [← h, skyscraper_presheaf_cocone_of_specializes_ι_app,
+    eq_to_hom_trans_assoc, eq_to_hom_refl, category.id_comp] }
+
+/--
+If `y ∈ closure {p₀}`, then the stalk of `skyscraper_presheaf p₀ S` at `y` is `S`.
+-/
+@[reducible]
+noncomputable def skyscraper_presheaf_stalk_of_specializes [has_colimits C]
+  {y : X} (h : p₀ ⤳ y) : (skyscraper_presheaf p₀ S).stalk y ≅ S :=
+colimit.iso_colimit_cocone ⟨_, skyscraper_presheaf_cocone_is_colimit_of_specializes p₀ S h⟩
+
+/--
+The cocone at `*` for the stalk functor of `skyscraper_presheaf p₀ S` when `y ∉ closure {p₀}`
+-/
+@[simps] def skyscraper_presheaf_cocone (y : X) :
+  cocone ((open_nhds.inclusion y).op ⋙ skyscraper_presheaf p₀ S) :=
+{ X := terminal C,
+  ι :=
+  { app := λ U, terminal.from _,
+    naturality' := λ U V inc, terminal_is_terminal.hom_ext _ _ } }
+
+/--
+The cocone at `*` for the stalk functor of `skyscraper_presheaf p₀ S` when `y ∉ closure {p₀}` is a
+colimit
+-/
+noncomputable def skyscraper_presheaf_cocone_is_colimit_of_not_specializes
+  {y : X} (h : ¬p₀ ⤳ y) : is_colimit (skyscraper_presheaf_cocone p₀ S y) :=
+let h1 := mem_nhds_of_not_specializes p₀ h in
+{ desc := λ c, eq_to_hom (if_neg h1.some_spec).symm ≫ c.ι.app (op h1.some),
+  fac' := λ c U, begin
+    change _ = c.ι.app (op U.unop),
+    simp only [← c.w (hom_of_le $ @inf_le_left _ _ h1.some U.unop).op,
+      ← c.w (hom_of_le $ @inf_le_right _ _ h1.some U.unop).op, ← category.assoc],
+    congr' 1,
+    refine ((if_neg _).symm.rec terminal_is_terminal).hom_ext _ _,
+    exact λ h, h1.some_spec h.1,
+  end,
+  uniq' := λ c f H, begin
+    rw [← category.id_comp f, ← H, ← category.assoc],
+    congr' 1, apply terminal_is_terminal.hom_ext,
+  end }
+
+/--
+If `y ∉ closure {p₀}`, then the stalk of `skyscraper_presheaf p₀ S` at `y` is `*`
+-/
+@[reducible]
+noncomputable def skyscraper_presheaf_stalk_of_not_specializes [has_colimits C]
+  {y : X} (h : ¬p₀ ⤳ y) : (skyscraper_presheaf p₀ S).stalk y ≅ terminal C :=
```

Comment:
I think it would be useful to have a variant stating `is_terminal ((skyscraper_presheaf p₀ S).stalk y)`.

Reply:
Since other instances of `terminal` are marked as `defs` instead of `instances`, I have added a definition as well.